src/utils/stream.rs

```diff
@@ -1,80 +1,111 @@
-use std::{io::{Write, stdout, stderr, Result}, time::Instant};
+use std::{
+    io::{stderr, stdout, Result, Write},
+    time::Instant,
+};
 
 /// Handles the Printing logic for the Spinner
-#[derive(Default, Copy, Clone)]
+#[derive(Default, Clone)]
 pub enum Stream {
     #[default]
     Stderr,
     Stdout,
+    #[cfg(feature = "ratatui")]
+    Ratatui(crate::SpinnerWidget),
 }
+
 impl Stream {
```

Comment:
There's a lot of churn in this. In summary:
- rearranged to read well top to bottom
- methods which were passed the writer now just get self and pull the writer if needed
- write calls which would have written ansi / control characters to the stream are now separated from the calls which write info (e.g. `\r` and the erase line sequence.